server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -86,6 +92,35 @@ final class FieldTypeLookup {
             }
         }
 
+        // Pass-though subfields can be referenced without the prefix corresponding to the
+        // PassThroughObjectMapper name. This is achieved by adding a second reference to their
+        // MappedFieldType using the remaining suffix.
+        Map<String, PassThroughObjectMapper> passThroughFieldAliases = new HashMap<>();
+        for (PassThroughObjectMapper passThroughMapper : passThroughMappers) {
+            for (Mapper subfield : passThroughMapper.mappers.values()) {
+                if (subfield instanceof FieldMapper fieldMapper) {
+                    String name = fieldMapper.simpleName();
+                    // Check for conflict between PassThroughObjectMapper subfields.
+                    PassThroughObjectMapper conflict = passThroughFieldAliases.put(name, passThroughMapper);
+                    if (conflict != null) {
+                        // Check the priorities of the conflicting objects.
+                        if (conflict.priority() > passThroughMapper.priority()) {
```

Comment:
let's reverse the priority? A field from a passthrough field with a higher priority wins.

Reply:
I think this is the case? If the inserted object has higher priority we leave it there, otherwise the conflicting one is put back. In both cases, `fullNameToFieldType` will contain the alias for the object with the highest priority?